pkg/binary/envoy/debug/pii_test.go

```diff
@@ -0,0 +1,132 @@
+	logWithOneField := `[2019-09-05T17:53:36.908Z]`
+
+	// test cases wrapped up in a map where key is the test case name and value is the struct
+	// defining the inputs and expecte outcome
+	tests := map[string]struct {
+		// input fields
+		in     []string
+		format string
+		pii    map[string]bool
+
+		// output fields
+		outLogs []string
+		outErr  string
+	}{
+		// positive test cases
+		// only return an array containing one valid log with [%START_TIME%] field hashed
+		"simple": {
+			in:     []string{validLog, logWithOneField},
+			format: format,
+			pii:    map[string]bool{"[%START_TIME%]": true},
+			// expected expression in "" replaced by ''
+			outLogs: []string{`pii 'GET /productpage HTTP/1.1' 200 - - 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko)` +
+				` Chrome/76.0.3809.132 Safari/537.36'`},
+			outErr: "",
+		},
+		"multiple_pii_fields": {
+			in:     []string{validLog, logWithOneField},
+			format: format,
+			pii: map[string]bool{
+				"[%START_TIME%]": true,
+				"%REQ(:METHOD)% %REQ(X-ENVOY-ORIGINAL-PATH?:PATH)% %PROTOCOL%": true,
+				"%RESPONSE_CODE%": true,
+			},
+			outLogs: []string{`pii pii pii - - 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.132 Safari/537.36'`},
+			outErr:  "",
+		},
+
+		// negative test cases
+		"empty_logs": {
+			in:      []string{},
+			format:  format,
+			pii:     map[string]bool{"[%START_TIME%]": true},
+			outLogs: []string{},
+			outErr:  "",
+		},
+		"all_invalid_logs": {
+			in:      []string{"test", "test1", "test2"},
+			format:  format,
+			pii:     map[string]bool{"[%START_TIME%]": true},
+			outLogs: []string{},
+			outErr:  "",
+		},
+		"empty_format_string": {
+			in:      []string{validLog, logWithOneField},
+			format:  "",
+			pii:     map[string]bool{"[%START_TIME%]": true},
+			outLogs: []string{},
+			outErr:  "",
+		},
+		"empty_pii_map": {
+			in:     []string{validLog, logWithOneField},
+			format: format,
+			pii:    map[string]bool{},
+			outLogs: []string{`'[2019-09-05T17:53:36.908Z]' 'GET /productpage HTTP/1.1' 200 - - 'Mozilla/5.0 (X11; Linux x86_64)` +
+				` AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.132 Safari/537.36'`},
+			outErr: "",
+		},
+		"invalid_pii_map": {
+			in:     []string{validLog, logWithOneField},
+			format: format,
+			pii:    map[string]bool{"test": true, "test2": true},
+			outLogs: []string{`'[2019-09-05T17:53:36.908Z]' 'GET /productpage HTTP/1.1' 200 - - 'Mozilla/5.0 (X11; Linux x86_64)` +
+				` AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.132 Safari/537.36'`},
+			outErr: "",
+		},
+	}
+
+	piiFunc := func(string) string { return "pii" }
+
+	// defining subtest with t.Run
+	//nolint:gocritic,scopelint
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			actualLogs, actualErr := ProcessLogs(tc.in, tc.format, tc.pii, piiFunc)
```

Comment:
`actual, err` - we know what's under test - the logs - and we know what `err` means - the error in this context. We don't need the extra context of `actual` in `actualErr` nor do we need the `Logs` suffix in `actualLogs` - both are still obvious in meaning.